Trimble.Modus.Components/Controls/TMInput/TMInput.cs

```diff
@@ -0,0 +1,871 @@
+        }
+
+        /// <summary>
+        /// Gets or sets the title text
+        /// </summary>
+        public string TitleText
+        {
+            get { return (string)GetValue(TitleTextProperty); }
+            set { SetValue(TitleTextProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the left icon in entry's source
+        /// </summary>
+        public ImageSource LeftIconSource
+        {
+            get { return (ImageSource)GetValue(LeftIconSourceProperty); }
+            set { SetValue(LeftIconSourceProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the right icon in entry's source
+        /// </summary>
+        public ImageSource RightIconSource
+        {
+            get { return (ImageSource)GetValue(RightIconSourceProperty); }
+            set { SetValue(RightIconSourceProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the command for left icon
+        /// </summary>
+        public ICommand LeftIconCommand
+        {
+            get => (ICommand)GetValue(LeftIconCommandProperty);
+            set => SetValue(LeftIconCommandProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the command parameter for left icon
+        /// </summary>
+        public object LeftIconCommandParameter
+        {
+            get => GetValue(LeftIconCommandParameterProperty);
+            set => SetValue(LeftIconCommandParameterProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the right icon command
+        /// </summary>
+        public ICommand RightIconCommand
+        {
+            get => (ICommand)GetValue(RightIconCommandProperty);
+            set => SetValue(RightIconCommandProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the right icon command parameter
+        /// </summary>
+        public object RightIconCommandParameter
+        {
+            get => GetValue(LeftIconCommandParameterProperty);
+            set => SetValue(LeftIconCommandParameterProperty, value);
+        }
+        #endregion
+
+        #region Property changes
+
+        /// <summary>
+        /// This method is triggered when the text in the input field is changed, 
+        /// this performs text validation to update the validation text field
+        /// </summary>
+        /// <param name="bindable">Object</param>
+        /// <param name="oldValue">Old value</param>
+        /// <param name="newValue">New value</param>
+        private static void OnInputTextChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            var tMinput = (TMInput)bindable;
+            if(tMinput.InputValidation != null)
+            {
+                if (!string.IsNullOrEmpty((string)newValue))
+                {
+                    var result = tMinput.InputValidation?.Invoke(tMinput);
+
+                    tMinput._validationLabel.Text = result.Item2;
```

Comment:
When the `result.Item2` is null or empty string, then the helper text should be displayed.

Reply:
Done